### ISSUE TITLE
Support for Ubuntu 20.04

### ISSUE THREAD

**Issue by kript** [irods/irods#4883]
## Feature

Can we get support for Ubuntu 20.04, now its been released please?
We will likely move next year from 18.04.

cheers!

John

**Comment by trel** [comment]
you're great.    :)

**Comment by your-highness** [comment]
Tried using bionic repository:
```bash
>sudo apt install -y irods-icommands
Reading package lists... Done
Building dependency tree
Reading state information... Done
Some packages could not be installed. This may mean that you have
requested an impossible situation or if you are using the unstable
distribution that some required packages have not yet been created
or been moved out of Incoming.
The following information may help to resolve the situation:

The following packages have unmet dependencies:
 irods-icommands : Depends: irods-runtime (= 4.2.8) but it is not going to be installed
                   Depends: libssl1.0.0 but it is not installable
```

`libssl1.0.0` in bionic is not available but `libssl1.1` is installed.

Any fix for this?

**Comment by trel** [comment]
Hi @your-highness, `libssl1.1` support is the same issue as https://github.com/irods/irods/pull/4999.

**Comment by your-highness** [comment]
Hi,

I would like to provide my approach to make irods-icommands available on Ubuntu 20.04 (focal) by using the iRODS bionic repos:

```bash
###
# 1. Prepare IROS APT repository
LSB_RELEASE="bionic" #Using bionic repos because focal is not available
wget -qO - https://packages.irods.org/irods-signing-key.asc | sudo apt-key add -
echo "deb [arch=amd64] https://packages.irods.org/apt/ ${LSB_RELEASE}  main" \
  | sudo tee /etc/apt/sources.list.d/renci-irods.list
sudo apt-get update

###
# 2. Install python-urlib3, python-requests and libssl1.0.0
wget -c \
  http://security.ubuntu.com/ubuntu/pool/main/p/python-urllib3/python-urllib3_1.22-1ubuntu0.18.04.2_all.deb \
  http://security.ubuntu.com/ubuntu/pool/main/r/requests/python-requests_2.18.4-2ubuntu0.1_all.deb \
  http://security.ubuntu.com/ubuntu/pool/main/o/openssl1.0/libssl1.0.0_1.0.2n-1ubuntu5.4_amd64.deb
sudo apt install \
  ./python-urllib3_1.22-1ubuntu0.18.04.2_all.deb \
  ./python-requests_2.18.4-2ubuntu0.1_all.deb \
  ./libssl1.0.0_1.0.2n-1ubuntu5.4_amd64.deb
rm -rf \
  ./python-urllib3_1.22-1ubuntu0.18.04.2_all.deb \
  ./python-requests_2.18.4-2ubuntu0.1_all.deb \
  ./libssl1.0.0_1.0.2n-1ubuntu5.4_amd64.deb

###
# 3. Install irods-icommands
sudo apt install -y irods-icommands
```

I hope this helps

**Comment by trel** [comment]
Nice - thank you.

**Comment by holtgrewe** [comment]
The last url in the wget line must now be http://security.ubuntu.com/ubuntu/pool/main/o/openssl1.0/libssl1.0.0_1.0.2n-1ubuntu5.5_amd64.deb

**Comment by HunterKing** [comment]
> The last url in the wget line must now be http://security.ubuntu.com/ubuntu/pool/main/o/openssl1.0/libssl1.0.0_1.0.2n-1ubuntu5.5_amd64.deb

They have since updated to 5.6, so the url to drop in is (and don't forget to change the apt install!):
http://security.ubuntu.com/ubuntu/pool/main/o/openssl1.0/libssl1.0.0_1.0.2n-1ubuntu5.6_amd64.deb

**Comment by gspeed0689** [comment]
It looks like they updated from 5.6 to 5.7, and changing the 6 to a 7 in the script worked in Ubuntu 20.04 WSL

**Comment by your-highness** [comment]
> 
> 
> It looks like they updated from 5.6 to 5.7, and changing the 6 to a 7 in the script worked in Ubuntu 20.04 WSL

Updated the commands above.

**Comment by your-highness** [comment]
Any plans to make https://packages.irods.org available for Ubuntu 20.04 (focal)?

**Comment by trel** [comment]
Yes, 4.3.0 packages will appear there.
